pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -251,71 +254,120 @@ func StartLibvirt(stopChan chan struct{}) {
 	}()
 }
 
-func StartVirtlog(stopChan chan struct{}, domainName string) {
-	go func() {
-		for {
-			var args []string
-			args = append(args, "-f")
-			args = append(args, "/etc/libvirt/virtlogd.conf")
-			// #nosec No risk for attacket injection. Args  are predefined strings
-			cmd := exec.Command("/usr/sbin/virtlogd", args...)
+func startVirtlogdLogging(stopChan chan struct{}, domainName string) {
+	for {
+		cmd := exec.Command("/usr/sbin/virtlogd", "-f", "/etc/libvirt/virtlogd.conf")
 
-			exitChan := make(chan struct{})
+		exitChan := make(chan struct{})
+
+		err := cmd.Start()
+		if err != nil {
+			log.Log.Reason(err).Error("failed to start virtlogd")
+			panic(err)
+		}
+
+		go func() {
+			logfile := fmt.Sprintf("/var/log/libvirt/qemu/%s.log", domainName)
 
-			err := cmd.Start()
+			// It can take a few seconds to the log file to be created
+			for {
+				_, err = os.Stat(logfile)
+				if !os.IsNotExist(err) {
+					break
+				}
+				time.Sleep(time.Second)
+			}
+			// #nosec No risk for path injection. logfile has a static basedir
+			file, err := os.Open(logfile)
 			if err != nil {
-				log.Log.Reason(err).Error("failed to start virtlogd")
-				panic(err)
+				errMsg := fmt.Sprintf("failed to open logfile in path: \"%s\". Error: %s", logfile, err.Error())
+				log.Log.Reason(err).Error(errMsg)
```

Comment:
wouldn't `Reason(err)` log the same error as err.Error()? are we logging it twice?

Reply:
Makes sense. I wasn't aware that that's what `Reason(err)` is doing. Thanks :)